src/app/utils/getRPCAccountBalances.ts

```diff
@@ -0,0 +1,47 @@
+import { Runtime, RuntimeSdkBalance } from '../../oasis-nexus/api'
+import * as oasis from '@oasisprotocol/client'
+import * as oasisRT from '@oasisprotocol/client-rt'
+import { Network } from '../../types/network'
+import { getTokensForScope, paraTimesConfig } from '../../config'
+import { fromBaseUnits } from './helpers'
+
+const grpcUrls: { [key in Network]: string } = {
+  [Network.mainnet]: 'https://grpc.oasis.dev',
+  [Network.testnet]: 'https://testnet.grpc.oasis.dev',
+}
+
+export async function getRPCAccountBalances(
+  oasisAddress: string,
+  scope: {
+    network: Network
+    layer: Runtime
+  },
+): Promise<RuntimeSdkBalance[]> {
+  const paratimeConfig = paraTimesConfig[scope.layer][scope.network]
+  if (!paratimeConfig.runtimeId) throw new Error('Paratime is not configured')
+
+  const nic = new oasis.client.NodeInternal(grpcUrls[scope.network])
+  const accountsWrapper = new oasisRT.accounts.Wrapper(oasis.misc.fromHex(paratimeConfig.runtimeId))
+
+  const rawBalances = (
+    await accountsWrapper
+      .queryBalances()
```

Comment:
I think native balance querying should be enough, as ERC-20 tokens for example are showing correct balances. But since it is going to be removed in the future, I guess it doesn't matter that much.